### PR TITLE
doc(all): Deprecate Java 7 support

### PR DESCRIPTION
**Pull request by timtay-microsoft**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -9,11 +9,13 @@ Due to security considerations, build logs are not publicly available.
...
 
 This repository contains the following:
-* **Azure IoT Hub device SDK for Java**: connect client devices to Azure IoT Hub (supports Java 7+)
-* **Azure IoT Hub service SDK for Java**: enables developing back-end applications for Azure IoT (supports Java 8+)
+* **Azure IoT Hub device SDK for Java**: connect client devices to Azure IoT Hub (supports Java 8)
+* **Azure IoT Hub service SDK for Java**: enables developing back-end applications for Azure IoT (supports Java 8)
 * **Azure IoT Device Provisioning device SDK for Java**: provision devices to Azure IoT Hub using Azure IoT Device Provisioning
 * **Azure IoT Device Provisioning service SDK for Java**: manage your Provisioning service instance from a back-end Java application
 
+Note: The Device Client library no longer supports Java 7. Customers are advised to upgrade to Java 8.
```
Comment (abhipsaMisra) [inline readme.md:17]: We should also list the last mvn package which did have Java 7 support.

Comment (timtay-microsoft) [reply readme.md:17]: Good idea

**Pull request closed without merging**